### ISSUE TITLE
Issue with KC705 and KCU105 PCIe FW

### ISSUE THREAD

**Issue by zixu** [slaclab/atlas-rd53-fmc-dev#5]
got the same issue for KC705 and KCU105 board with latest release Release v4.3.2

Follow the instruction here: https://github.com/slaclab/atlas-rd53-fmc-dev
![image](https://user-images.githubusercontent.com/4989435/97064215-2b708880-1559-11eb-816f-b6ae407e0d26.png)

PCIe firmware is transferred from PC to the Evaluation board through JTag cable by Vivado. 
PCIe HW is founded: 
![image](https://user-images.githubusercontent.com/4989435/97064152-cddc3c00-1558-11eb-9bf6-8d927bbbfa58.png)

compile the latest master branch driver. After load the driver, the /dev/datadev_0 presented, 
![image](https://user-images.githubusercontent.com/4989435/97064232-4a6f1a80-1559-11eb-9fe6-0562d853b75a.png)
![image](https://user-images.githubusercontent.com/4989435/97064253-75f20500-1559-11eb-8bd1-40dd6e630251.png)

but **cat** failed
![image](https://user-images.githubusercontent.com/4989435/97064274-9621c400-1559-11eb-9638-7dfbf1abde95.png)

and nothing output with dma app:
![image](https://user-images.githubusercontent.com/4989435/97064811-b2bffb00-155d-11eb-9cb2-617acbdf094e.png)

![image](https://user-images.githubusercontent.com/4989435/97064816-bd7a9000-155d-11eb-8be1-644a99343587.png)

**run the pyrogue gui script, got time out issue**
![image](https://user-images.githubusercontent.com/4989435/97064325-e0a34080-1559-11eb-9b2b-f30940e606ac.png)

![image](https://user-images.githubusercontent.com/4989435/97064355-1f38fb00-155a-11eb-89a7-f30b1e773bff.png)

Also tested some different version of aes driver, like v5.7.0, v5.3.1, v5.0.0, v2.2.0. None of them works.

**Comment by zixu** [comment]
@ruck314  Any specific jumper needs to set?

**Comment by zixu** [comment]
BTW, the PC is using PCIe Gen2 x16

**Comment by ruck314** [comment]
First mistake is that you type the "cat" command wrong. It should be `cat /proc/data_dev0` and not `cat /dev/data_dev0`
```bash
# Check for the loaded device
$ cat /proc/data_dev0
```
Next is that DMA app that you tried to run will only work if you have loopback firmware.  If you are using the RD53 FW, then that `stock` DMA applications will not work.

Please send me the following output for the following commands:
```bash
$ cat /proc/data_dev0
$ sudo lspci -s 02:00.0 -vvv
```

**Comment by zixu** [comment]
![image](https://user-images.githubusercontent.com/4989435/97067971-08a19c80-1578-11eb-81dd-170d234069e8.png)

![image](https://user-images.githubusercontent.com/4989435/97067979-1a833f80-1578-11eb-81a9-8118bddbb6fd.png)

**Comment by ruck314** [comment]
Looks like the link only locked to PCIe GEN2 x 2 lanes.  KCU705 support up to GEN2 and 8 lanes. You should double check that J32 has pin5 and6 jumpered:
![image](https://user-images.githubusercontent.com/11821758/97068213-8d8db580-157a-11eb-98bc-63b6c97de4d4.png)
Can you send me the `dmesg` log?

**Comment by zixu** [comment]
For KCU105:

I confirm that I connected the pin 5, 6 of Jumper 74, 
![image](https://user-images.githubusercontent.com/4989435/97069855-ea439d00-1587-11eb-8427-c0ac99c0d229.png)

![image](https://user-images.githubusercontent.com/4989435/97069747-1f032480-1587-11eb-96fd-3752a6af0f41.png)

![image](https://user-images.githubusercontent.com/4989435/97069756-36421200-1587-11eb-8e03-8f7c0a9542d0.png)

dmesg output:
https://1drv.ms/u/s!AhEfroiycPJpguscfibTGlrhQqQMgg?e=fJWD7y

**Comment by zixu** [comment]
For KC705
![image](https://user-images.githubusercontent.com/4989435/97070307-2ed13780-158c-11eb-9121-d42a4a115cfa.png)

![image](https://user-images.githubusercontent.com/4989435/97070321-3db7ea00-158c-11eb-9fe2-0c0e55cf1cfb.png)

dmesg output
https://1drv.ms/u/s!AhEfroiycPJpgusdzUfSQixDaiq6XQ?e=wchY5n

**Comment by zixu** [comment]
Jumper setting:

KCU105
![image](https://user-images.githubusercontent.com/4989435/97070760-c5532800-158f-11eb-83ae-c39ed047aedf.png)


KC705
![image](https://user-images.githubusercontent.com/4989435/97070769-d4d27100-158f-11eb-902d-d9a4e75d8eaf.png)

**Comment by ruck314** [comment]
What is the Linux distro and Linux kernel version that you are using?

**Comment by zixu** [comment]
Linux distro: CentOS 7.6
Kernel: 3.10.0-957.e17.x86_64

**Comment by zixu** [comment]
Tested the KCU105 with the v4.4 FW, and aes v5.14.3. Issue still existed

![image](https://user-images.githubusercontent.com/4989435/97243535-bfd62780-17b3-11eb-9527-bd929ff4f1a2.png)


![image](https://user-images.githubusercontent.com/4989435/97243693-1e9ba100-17b4-11eb-8113-1d50c8ba5bcc.png)


dmesg log: https://1drv.ms/u/s!AhEfroiycPJpgushjNGD983mh6r5tw?e=Cx1gS7

**Comment by ruck314** [comment]
This issue resolved in v4.4.1 tag release.
https://github.com/slaclab/atlas-rd53-fmc-dev/releases/tag/v4.4.1
